### ISSUE TITLE
Enable building only some outputs of conda recipes

### ISSUE THREAD

**Issue by vyasr** [rapidsai/build-planning#61]
This request comes from https://github.com/rapidsai/cudf/issues/11119. The desire is to be able to build specific outputs of a multi-output recipe while skipping others.

I don't know if this is fundamentally something RAPIDS should be handling or if it is more of a request for conda itself, but it's something to consider. We should also see what impacts #47 would have on this. I don't know what the current state of support for multi-output recipes is in rattler-build, but it appears that there is at least some support for it already (see [this example](https://github.com/prefix-dev/rattler-build/blob/main/examples/mamba/recipe.yaml)). Whether it covers all the use cases we need, and whether it would help enabling this particular request, remains to be seen.

**Comment by jakirkham** [comment]
Yeah would recommend raising a conda-build issue or in the case of #47 a rattler-build issue

**Comment by vyasr** [comment]
Now that we are actively moving towards rattler-build, I opened https://github.com/prefix-dev/rattler-build/issues/1425 to discuss this. We are not going to implement this ourselves, so I'm going to close this issue now.

**Comment by gforsyth** [comment]
Yeah, I can do that!

**Comment by gforsyth** [comment]
Tested this locally on the `rmm` recipe and I can confirm that you can output subsets, but I don't think you can output individual outputs, unless that output happens to be the first in the list of outputs.

E.g. with `librmm`, there is `librmm` and `librmm-tests` and it is possible to only build `librmm` since it is defined first, via `--up-to librmm` but you couldn't only build `librmm`, since `--up-to librmm-tests` would also include `librmm`

**Comment by vyasr** [comment]
I think that is a sensible requirement since it doesn't really make sense to build an output that has a pinning to a different output without building the second output first. Not breaking the dependency graph seems like a good thing to me. I'll close this issue again and update the corresponding other issues. Thanks Gil!
